target/ppc/fpu_helper.c

```diff
@@ -462,6 +477,38 @@ void helper_fpscr_check_status(CPUPPCState *env)
     }
 }
 
+static void update_fpscr(CPUPPCState *env, int mask)
+{
+    int flags = get_float_exception_flags(&env->fp_status);
+
+    if ((mask & FP_VXSNAN) && (flags & float_flag_invalid_snan)) {
+        float_invalid_op_vxsnan(env, GETPC());
```

Comment:
Changed this in [d1450ba](https://github.com/PPC64/qemu/pull/76/commits/d1450baa379b2decef9aa7ce7108ea1b3e84d884). Could you take a look and see if the new approach is better?

Reply:
It seems better. I guess we can open `update_fpscr` in `finish_fp_operation` since there isn't any other caller. The same should happen with `raise_excp` when we replace `do_float_check_status` completely.

Reply:
Changed in [411bc74](https://github.com/PPC64/qemu/pull/76/commits/411bc74e53d0bfc77b0d68e0eb73b406f8c473ef) and made `raise_excp` inline

Reply:
I'd remove the inline from `raise_excp`, the compiler should be smart enough to decide whether it's better to inline, tail-call, etc.